src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/TransportFactory.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\MongoDB\Transport;
+
+use MongoDB\Database;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Messenger\Bridge\MongoDB\Extension\DocumentEnhancer;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+
+/**
+ * @template-implements TransportFactoryInterface<MongoDbUnresettableTransport>
+ */
+final class TransportFactory implements TransportFactoryInterface
+{
+    private const DEFAULT_OPTIONS = [
+        self::COLLECTION_NAME => 'messenger_messages',
+        self::QUEUE_NAME => 'default',
+        self::REDELIVER_TIMEOUT => 3_600,
+        self::DOCUMENT_ENHANCERS => [],
+        self::RESETTABLE => true,
+    ];
+
+    public const CONNECTION_NAME = 'connection_name';
+
+    public const COLLECTION_NAME = 'collection_name';
+
+    public const QUEUE_NAME = 'queue_name';
+
+    public const REDELIVER_TIMEOUT = 'redeliver_timeout';
+
+    public const DOCUMENT_ENHANCERS = 'document_enhancers';
+
+    public const RESETTABLE = 'resettable';
+
+    public function __construct(private readonly ContainerInterface $container)
+    {
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     */
+    public function supports(string $dsn, array $options): bool
+    {
+        return str_starts_with($dsn, 'facile-it-mongodb://');
```

Comment:
Do we want to support any [MongoDB connection](https://www.mongodb.com/docs/manual/reference/connection-string/) string here?
```suggestion
        return str_starts_with($dsn, 'mongodb://') || str_starts_with($dsn, 'mongodb+srv://');
```

Reply:
That would be nice! Maybe in a separate PR?

Reply:
We cannot keep `facile-it-mongodb` as scheme name, it will conflict with your bundle.